lib/bitpagos/client.rb

```diff
@@ -51,10 +51,20 @@ def transaction_count
 
     private
 
+    # Hits the Bitpagos transaction API, returns a hash with results
+    #
+    # @param [String] State (Pending, Waiting, Completed, Partially Paid)
+    # @param [String] Transaction ID
+    # @return [Hash]
     def retrieve_transactions(query = nil, transaction_id = nil)
+      headers.merge!(params: { status: query }) if query
       url = "#{API_BASE}/transaction/#{transaction_id}"
-      response = RestClient.get(url, headers)
-      JSON.parse(response)
+      begin
+        response = RestClient.get(url, headers)
+        JSON.parse(response)
+      rescue RestClient::Unauthorized => error
+        raise Bitpagos::Errors::Unauthorized.new(error.message)
```

Comment:
@mauro-oto The problem with this implementation is that it's losing the `cause` of the problem. Maybe you could take advantage of this? http://ruby-doc.org/core-2.2.3/Exception.html#method-i-cause